js/app.js

```diff
@@ -0,0 +1,104 @@
+import todoItem from './todoItem.js';
+import filterItem from './filterItem.js';
+let ID = 0;
+let DATA = []; // {id, context, complete}
+let HASH = "all";
+
+
+
+/************* element select & event setting ****************/ 
+const $todoInputElement = document.getElementById('new-todo-title');
+const $todoListElement = document.getElementById('todo-list');
+const $filtersElement = document.querySelector('.filters');
+const $countElement = document.querySelector('.todo-count strong')
+
+$todoInputElement.addEventListener('keyup', (event) => {
+	const key = event.key;
+	const valueTrim = event.target.value.trim()
+
+    if (key === "Enter" && valueTrim) {
+        let data = {
+            id: ID++,
+            context: valueTrim,
+            complete: false
+		}
+		inputEvent(data)
+		render()
+		$todoInputElement.value = '';
+    }
+})
+$todoListElement.addEventListener('keyup', (event) => {
+	const key = event.key;
+	const target = event.target;
+	const parent = target.parentElement;
+	const valueTrim = event.target.value.trim()
+	let labelElement = parent.querySelector('label')
+
+	if (key === 'Escape') {
+		parent.classList.remove('editing')
+		target.value = labelElement.innerHTML
+	} else if ( key === 'Enter' && valueTrim) {
+		let index = parent.id.split('-')[1]
+		DATA[index].context = valueTrim
+		render()
+	}
+
+})
+// toggle change event 
+$todoListElement.addEventListener('change', (event) => {
+	const target = event.target;
+	const targetId = target.parentElement.id
+
+	if (target.id) {
+		DATA.forEach((v, i, a) => {
+			if ( `item-${v.id}` === targetId ) {
+				v.complete = !v.complete;
+			}
+		})
+		render();
+	}
+})
+// button click event
+$todoListElement.addEventListener('click', (event) => {
+	const target = event.target
+	const targetId = target.parentElement.id
+	
+	if (target.className === 'destroy') {
+		DATA = DATA.filter((v) => {
+			return ( `item-${v.id}` === targetId ) ? false : true;
+		})
+		render();
+	}
+})
+
+
+// 입력된 값 수저하는 기능
+$todoListElement.addEventListener('dblclick', (event) => {
+	const target = event.target 
+	const parent = target.parentElement
+	parent.classList.add('editing')
+})
+window.addEventListener('hashchange', (event) => {
+	HASH = event.newURL.split('#')[1] || 'all'
+	render();
+})
+
+/**************** App logic ****************/
+const inputEvent = (newData) => {
+	DATA.push(newData)
+}
+const render =  () => {
+	let filteredData = DATA.filter((v) => {
+		if (HASH === 'completed') {
+			return (v.complete) ? true : false
+		} else if (HASH === 'active') {
+			return !(v.complete) ? true : false
+		} else {
+			return true
+		}
+	})
```

Comment:
```suggestion
        const filteredData = DATA.filter(({ completed }) => (HASH === 'all') ||
                                                            (HASH === 'completed' && completed) ||
                                                            (HASH === 'active' && !completed));
```
이런식으로 표현할 수 있습니다.
일단 변수는 무조건 const로 선언한다고 생각해주세요. 보통 변수에 저장한 값이 변하는 경우는 거의 없습니다.
배열이나 객체 또한 변수를 직접적으로 변경하는게 아니라 변수가 바라보고 있는 주소의 값이 변경되는 것이기 때문에 const로 해줘야합니다.

Reply:
적어주신 방법으로 조건식을 표현하는 것이 더 잘 읽혀서 좋습니다. 
좋은 방법 알려주셔서 감사합니다.